README.md

```diff
@@ -57,17 +83,82 @@ In order to upload the file, initiating microservice or the client should send t
 </form>
 ```
 
-# Error handling
+You can also post this form programmatically from your backend code or make an async call using AJAX. 
+It's important that the form will use multipart encoding (multipart/form-data), not application/x-www-form-urlencoded.
+If you use application/x-www-form-urlencoded, AWS will be returning confusing error messages.
+It's also important that the 'file' field of the form is the last one.
+
+When form will be submitted succesfully, the service will return HTTP 204 response with empty body. In case of any error,
+the service will return proper HTTP error response (4xx, 5xx) with details of the problem encoded as XML document. See Error handling
+section for details.
+
+When file is successfully uploaded by the user and checked by antivirus scanner, upscan service
+will make a POST call to the URL specified as a 'callbackUrl' parameter with the following body:
+
+```
+{
+    "reference" : "11370e18-6e24-453e-b45a-76d3e32ea33d",
+    "fileStatus" : "READY",
+    "url" : "https://bucketName.s3.eu-west-2.amazonaws.com?1235676"
+}
+```
+
+If the virus has been detected in the file or there is any other problem with the file. The following callback is being
+sent:
```

Comment:
"If a virus has been detected in the file or there is any other problem with the file, the following JSON is posted to the callback URL:"